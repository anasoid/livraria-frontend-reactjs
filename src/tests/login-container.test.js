import React from 'react';
import renderer from 'react-test-renderer';
import {shallow} from 'enzyme';
import LoginContainer from '../components/login/login-container';

describe('test LoginContainer component', () => {

  const wrapper = shallow(<LoginContainer />);

  it('renders without crashing', () => {
    shallow(<LoginContainer />);
    const tree = renderer.create(<LoginContainer />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('LoginContainer should be contain form', () => {
    expect(wrapper.find('form')).toExist();
  });

  it('LoginContainer should be contain input loginEmail', () => {
    const email = wrapper.find('#loginEmail');
    expect(email).toMatchSelector('input');
    expect(email).toHaveProp('name', 'email');
  });

  it('LoginContainer should be contain input loginSenha', () => {
    const senha = wrapper.find('#loginSenha');
    expect(senha).toMatchSelector('input');
    expect(senha).toHaveProp('name', 'senha');
  });

  it('LoginContainer should be contain button Efetue Login and tyoe submit', () => {
    const button = wrapper.find('button');
    expect(button).toHaveText('Efetue Login');
  })

});
