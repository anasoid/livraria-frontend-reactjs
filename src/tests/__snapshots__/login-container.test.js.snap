// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test LoginContainer component renders without crashing 1`] = `
<div>
  <span
    id="messages"
  />
  <form>
    <fieldset>
      <legend>
        Login
      </legend>
      <label
        htmlFor="loginEmail"
      >
        Email:
        <span>
          *
        </span>
      </label>
      <input
        id="loginEmail"
        name="email"
        type="email"
      />
      <br />
      <label
        htmlFor="loginSenha"
      >
        Senha:
        <span>
          *
        </span>
      </label>
      <input
        id="loginSenha"
        name="senha"
        type="password"
      />
      <br />
      <button
        type="submit"
      >
        Efetue Login
      </button>
    </fieldset>
  </form>
</div>
`;
